cli/lib/kontena/cli/stacks/upgrade_command.rb

```diff
@@ -219,6 +220,7 @@ def run_upgrades(changes)
     def run_deploys(deployable_stacks)
       deployable_stacks.each do |deployable_stack|
         Kontena.run!(['stack', 'deploy', deployable_stack])
+        set_env_variables(stack_name, current_grid)
```

Comment:
Probably also not necessary? I don't think the `kontena stack deploy` command mutates the `ENV`?